src/Uno.UI/UI/Xaml/ResourceDictionary.cs

```diff
@@ -253,11 +253,9 @@ private void Set(in ResourceKey resourceKey, object value, bool throwIfPresent)
 			else
 			{
 				_values[resourceKey] = value;
-				if (value is ResourceDictionary)
-				{
-					ResourceDictionaryValueChange?.Invoke(this, EventArgs.Empty);
-				}
 			}
+
+			ResourceDictionaryValueChange?.Invoke(this, EventArgs.Empty);
```

Comment:
@davidjohnoliver This was a bug in my previous caching fix. It wasn't accounting for lazily initialized resources.

Reply:
![image](https://user-images.githubusercontent.com/31348972/136084779-69f22610-03a2-4d63-8f95-e6d107a55287.png)

@davidjohnoliver Here are actual numbers for both code paths on Skia SamplesApp when. It's almost 1.7x. But I'm not sure if that's a much of performance cost, nor I do have any other solution. I'll move this alone to another PR with fix if you think the fix is appropriate.

Reply:
Shouldn't `oldCounter++` be wrapped in `if (value is ResourceDictionary)`?

Reply:
@davidjohnoliver There is no `if (value is ResourceDictionary)` anywhere. It happens in the "else" path. This is where I previously was firing the event:

![image](https://user-images.githubusercontent.com/31348972/136087780-6ff24f87-ee96-4d0e-9168-2844be25d6ee.png)

Reply:
> @davidjohnoliver There is no `if (value is ResourceDictionary)` anywhere. It happens in the "else" path. This is where I previously was firing the event:
> 

Per the screenshot, you were previously firing the event inside `if (value is ResourceDictionary)`...

Reply:
Oops. Yeah yeah sorry. 😄

Reply:
@davidjohnoliver On regular app startup. It looks like `if (value is ResourceDictionary)` is never hit. So the event will indeed be fired much much more frequently. But I can't really think of other way to fix this. Any ideas?

Reply:
Well got an idea. Will separate it into another PR and ping you for review again :)